src/classes/CRLP_Batch_Base.cls

```diff
@@ -326,6 +335,226 @@ public abstract class CRLP_Batch_Base {
+                    whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
+                } else {
+                    whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+                }
+            } else if (maxDaysAgo != null) {
+                whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
+            } else if (maxYearsAgo != null) {
+                whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+            }
+        }
+        return whereClause += ')';
+    }
+
+    /**
+     * @description The max Integer for all active time-bound Days_Back rollups,
+     * or null if there are no active time-bound Days_Back rollups for this summary object
+     */
+    private Integer maxDaysAgo {
+        get {
+            if (maxDaysAgo == null) {
+                maxDaysAgo = getMaxIntegerValueFromRollupsForType(CRLP_Operation.TimeBoundOperationType.Days_Back);
+            }
+            return maxDaysAgo;
+        }
+        private set;
+    }
+
+    /**
+     * @description The max Integer for all active time-bound Years_Ago rollups,
+     * or null if there are no active time-bound Years_Ago rollups for this summary object
+     */
+    private Integer maxYearsAgo {
+        get {
+            if (maxYearsAgo == null) {
+                maxYearsAgo = getMaxIntegerValueFromRollupsForType(CRLP_Operation.TimeBoundOperationType.Years_Ago);
+            }
+            return maxYearsAgo;
+        }
+        private set;
+    }
+
+    /**
+     * @description Returns the maximum integer for time-bound rollups by time-bound type
+     * @param timeBoundType Time-bound type for which the max integer should be calculated
+     * @return max integer for the specified time bound type
+     */
+    private Integer getMaxIntegerValueFromRollupsForType(CRLP_Operation.TimeBoundOperationType timeBoundType) {
+        Integer maxInteger = null;
+        for (Rollup__mdt rollup : statefulCacheOfRollupsToBeProcessed) {
+            if (rollup.Active__c == true
+                    && String.isNotBlank(rollup.Time_Bound_Operation_Type__c)
+                    && rollup.Time_Bound_Operation_Type__c == timeBoundType.name()
+                    && rollup.Integer__c != null) {
+                if (maxInteger == null || rollup.Integer__c > maxInteger) {
+                    maxInteger = (Integer) rollup.Integer__c;
+                }
+            }
+        }
+        return maxInteger;
+    }
+
+    /**
+     * @description Determines whether the rollup should be run in incremental mode
+     * @param jobType the type of rollup job
+     * @return true if the rollup should be run in incremental mode, otherwise false
+     */
+    private Boolean getIsIncrementalModeEnabled() {
+        if (isScheduledJob == null || isScheduledJob == false) {
+            return false;
+        }
+
+        switch on (jobMode) {
+
+            when NonSkewMode {
+
+                Customizable_Rollup_Settings__c customizableRollupSettings =
```

Comment:
@kslewis  The assignment for `customizableRollupSettings` can be above the first `switch`.

Reply:
Ok, updated!